app/src/openxr/cpp/OpenXRInputSource.cpp

```diff
@@ -565,25 +571,74 @@ void OpenXRInputSource::EmulateControllerFromHand(device::RenderMode renderMode,
     for (int i = 0; i < mHandJoints.size(); i++) {
         vrb::Matrix transform = XrPoseToMatrix(mHandJoints[i].pose);
         if (mHandJoints[i].locationFlags & XR_SPACE_LOCATION_POSITION_VALID_BIT) {
-            float radius = mHandJoints[i].radius;
-            vrb::Matrix scale = vrb::Matrix::Identity().ScaleInPlace(vrb::Vector(radius, radius, radius));
-            transform.PostMultiplyInPlace(scale);
-
             if (renderMode == device::RenderMode::StandAlone)
                 transform.TranslateInPlace(kAverageHeight);
         } else {
             // This effectively hides the joint.
             transform.ScaleInPlace(vrb::Vector(0.0f, 0.0f, 0.0f));
         }
 
-        memcpy(&jointTransforms[i], &transform, sizeof(vrb::Matrix));
+        jointTransforms[i] = transform;
+    }
+
+    // Check whether palm of left hand is facing the head, to show navigate-back button
+    bool leftPalmFacesHead = false;
+    if (mHandeness == OpenXRHandFlags::Left) {
+#if defined(OCULUSVR)
+        if (mHandJoints[XR_HAND_JOINT_PALM_EXT].locationFlags & XR_SPACE_LOCATION_POSITION_VALID_BIT)
+            leftPalmFacesHead = !mHasAimState;
+#else
+        if (mHandJoints[XR_HAND_JOINT_WRIST_EXT].locationFlags & XR_SPACE_LOCATION_POSITION_VALID_BIT) {
+            vrb::Vector vector = vrb::Vector(0.0f, 0.0f, 1.0f);
+            vrb::Matrix palmMatrix = jointTransforms[XR_HAND_JOINT_WRIST_EXT];
+            vrb::Matrix headPalmMatrix = head.Inverse().PostMultiply(palmMatrix);
+            vrb::Vector diffVector = headPalmMatrix.MultiplyPosition(vector).Normalize();
+            leftPalmFacesHead = diffVector.z() >= kPalmHeadRotationZThreshold;
+        }
+#endif
+    }
+
+    // Scale joints according to their radius (for rendering)
```

Comment:
hmm seems unrelated to the patch?

Reply:
This scaling was part of the previous loop over joints but I had to move it here because to calculate whether palm faces head I need the joint transforms before scaling.